src/app/clusters/operational-credentials-server/operational-credentials-server.cpp

```diff
@@ -164,134 +164,13 @@ CHIP_ERROR OperationalCredentialsAttrAccess::Read(const ConcreteAttributePath &
-    status = writeFabricAttribute((uint8_t *) fabricDescriptor, static_cast<int32_t>(index));
-    chip::Platform::Delete(fabricDescriptor);
-    return status;
-}
-
-CHIP_ERROR writeFabricsIntoFabricsListAttribute()
+void fabricListChanged()
 {
-    emberAfPrintln(EMBER_AF_PRINT_DEBUG, "OpCreds: Call to writeFabricsIntoFabricsListAttribute");
-    CHIP_ERROR err = CHIP_NO_ERROR;
-
-    // Loop through fabrics
-    uint8_t fabricIndex = 0;
-    for (auto & fabricInfo : Server::GetInstance().GetFabricTable())
-    {
-        NodeId nodeId        = fabricInfo.GetPeerId().GetNodeId();
-        uint64_t fabricId    = fabricInfo.GetFabricId();
-        uint16_t vendorId    = fabricInfo.GetVendorId();
-        CharSpan fabricLabel = fabricInfo.GetFabricLabel();
-
-        // Skip over uninitialized fabrics
-        if (nodeId == kUndefinedNodeId)
-        {
-            emberAfPrintln(EMBER_AF_PRINT_DEBUG,
-                           "OpCreds: Skipping over uninitialized fabric with fabricId 0x" ChipLogFormatX64
-                           ", nodeId 0x" ChipLogFormatX64 " vendorId 0x%04" PRIX16,
-                           ChipLogValueX64(fabricId), ChipLogValueX64(nodeId), vendorId);
-            continue;
-        }
-        else if (writeFabric(fabricInfo.GetFabricIndex(), fabricId, nodeId, vendorId, fabricLabel, fabricInfo.GetRootPubkey(),
-                             fabricIndex) != EMBER_ZCL_STATUS_SUCCESS)
-        {
-            emberAfPrintln(EMBER_AF_PRINT_DEBUG,
-                           "OpCreds: Failed to write fabric with fabricId 0x" ChipLogFormatX64 " in fabrics list",
-                           ChipLogValueX64(fabricId));
-            err = CHIP_ERROR_PERSISTED_STORAGE_FAILED;
-            break;
-        }
-        fabricIndex++;
-    }
-
-    // Store the count of fabrics we just stored
-    emberAfPrintln(EMBER_AF_PRINT_DEBUG, "OpCreds: Stored %" PRIu8 " fabrics in fabrics list attribute.", fabricIndex);
-    uint16_t u16Index = fabricIndex;
-    if (writeFabricAttribute(reinterpret_cast<uint8_t *>(&u16Index)) != EMBER_ZCL_STATUS_SUCCESS)
-    {
-        emberAfPrintln(EMBER_AF_PRINT_DEBUG, "OpCreds: Failed to write fabric count %" PRIu8 " in fabrics list", fabricIndex);
-        err = CHIP_ERROR_PERSISTED_STORAGE_FAILED;
-    }
-
-    if (err == CHIP_NO_ERROR && Attributes::CommissionedFabrics::Set(0, fabricIndex) != EMBER_ZCL_STATUS_SUCCESS)
-    {
-        emberAfPrintln(EMBER_AF_PRINT_DEBUG, "OpCreds: Failed to write fabrics count %" PRIu8 " in commissioned fabrics",
-                       fabricIndex);
-        err = CHIP_ERROR_PERSISTED_STORAGE_FAILED;
-    }
-
-    if (err == CHIP_NO_ERROR && Attributes::SupportedFabrics::Set(0, CHIP_CONFIG_MAX_DEVICE_ADMINS) != EMBER_ZCL_STATUS_SUCCESS)
-    {
-        emberAfPrintln(EMBER_AF_PRINT_DEBUG, "OpCreds: Failed to write %d in supported fabrics count attribute",
-                       CHIP_CONFIG_MAX_DEVICE_ADMINS);
-        err = CHIP_ERROR_PERSISTED_STORAGE_FAILED;
-    }
```

Comment:
These bits still have to happen, no?  `SupportedFabrics` can get set in the initial endpoint-independent bringup, but `CommissionedFabrics` needs to be updated any time the fabric list changes.....

Reply:
Where we use it since these value are constructed in runtime during read. Do we have any case read them from ember attribute store directly?

Reply:
That's a good catch. Can we use `FabricTable:: FabricCount()` instead of writing this to the attribute store?

Reply:
For read? I think we have already done this. Is it not enough?

CHIP_ERROR OperationalCredentialsAttrAccess::ReadSupportedFabrics(EndpointId endpoint, AttributeValueEncoder & aEncoder)
{
    uint8_t fabricCount = CHIP_CONFIG_MAX_DEVICE_ADMINS;

    return aEncoder.Encode(fabricCount);
}

CHIP_ERROR OperationalCredentialsAttrAccess::ReadCommissionedFabrics(EndpointId endpoint, AttributeValueEncoder & aEncoder)
{
    return aEncoder.Encode(Server::GetInstance().GetFabricTable().FabricCount());
}

Reply:
>  I think we have already done this.

Ah, that wasn't in my local master when I was reading.  ;)

So ok, we don't need to write these to the store.  But we _do_ need to mark CommissionedFabrics dirty, right?

Reply:
In general, any time we add things to AttrAccess that can actually change we need to insert the corresponding `MatterReportingAttributeChangeCallback` calls.